### DISCUSSION TITLE
How does Neutralino Communicate with the frontend?

### DISCUSSION THREAD

**Discussion by JenuelDev** [neutralinojs/neutralinojs#571] [Q&A]
Hi! I was wondering because in Electron we have this ipcMain and ipcRenderer to communicate with the frontend and the background process. How does communication done in Neutralino Js? thanks thanks :)

**Comment by shalithasuranga** [comment]
Hello.. the client library communicates via a RESTful API. This is chosen to create multiple modes such as window, chrome, cloud, browser etc, You can check the source code of the router for the implementation: https://github.com/neutralinojs/neutralinojs/blob/main/server/router.cpp

Also, this is the HTTP server implementation; https://github.com/neutralinojs/neutralinojs/blob/main/server/neuserver.cpp

**Reply by JenuelDev** [reply]
is there a tutorial how to do this... thanks thanks :)

**Reply by JenuelDev** [reply]
Have you tried electrons ipcMain and ipcRenderer,, Im looking how to make something like it in neutralinojs

**Reply by shalithasuranga** [reply]
Ah Neutralinojs has a bit similar API https://neutralino.js.org/docs/api/events . But it is limited to some predefined native events. Are you trying to implement custom native code for an app?
